syntax = "proto3";

service Connector {
    rpc FetchProfiles (ProfilesRequest) returns (ProfilesReply) {}
    rpc SearchProfiles (ProfilesSearchRequest) returns (ProfilesReply) {}
}

message ProfilesRequest {
    optional string status = 1;
    int32 page = 2;
    int32 page_size = 3;
}

message ProfilesSearchRequest {
    string value = 1;
    optional string status = 2;
    int32 page = 3;
    int32 page_size = 4;
}

message ProfilesReply {
    message ProfilePhoto {
        string photo_id = 2;
        string url = 3;
    }
    message Profile {
        int32 age = 1;
        string bio = 2;
        string birth_date = 3;
        string city = 4;
        string created = 5;
        float distance = 6;
        int32 id = 7;
        bool liked = 8;
        string name = 9;
        repeated ProfilePhoto photos = 10;
        int64 s_number = 11;
        bool scheduled = 12;
        string user_id = 13;
    }
    message Reply {
        repeated Profile profiles = 1;
        int32 total = 2;
    }
    Reply reply = 1;
}